### DISCUSSION TITLE
Tokio MPSC Channel with Hyper StreamBody

### DISCUSSION THREAD

**Discussion by DreamTexX** [hyperium/hyper#3176] [Q&A] [answered]
Can someone give me an example on how to use the StreamBody with the mspc::channel from Tokio? I just cannot figure it out

**Comment by seanmonstar** [comment, marked as answer]
What version are you trying to use? What code have you tried so far?

**Reply by DreamTexX** [reply]
Hey, i use hyper version 1.0.0-rc.3 and http-body-util version 0.1.0-rc.2 and tried this code:

```rust
http1::Builder::new()
    .serve_connection(
        socket,
        service_fn(move |r: hyper::Request<Incoming>| async move {
            let (rx, tx) = mpsc::channel(128);
            
            let stream = ReceiverStream::new(tx);
            let body = StreamBody::new(stream);
  
            Ok(Response::new(body))
        }),
    )
    .await
    .expect("Cannot handle connection");
```
This kinda works, i would need to explicitly give the type and thats were i'm stuck... i actually have no clue what type the channel should have. I thought the `StreamBody` can be a wrapper around something like a `Vec<u8>` but it expects some type like `Result<Frame<???>, Error>`.

**Reply by seanmonstar** [reply]
Yea, so you just need to send `Result<Frame<Bytes>, SomeErrorType>`. If you stream will never error, you can make that type `std::convert::Infallible`.

**Reply by DreamTexX** [reply]
I'll try that, thanks

**Reply by DreamTexX** [reply]
Hey, this works. Thank you 😄
